traits/observers/tests/test_set_item_observer.py

```diff
@@ -0,0 +1,223 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import unittest
+from unittest import mock
+
+from traits.has_traits import HasTraits
+from traits.observers._set_item_observer import SetItemObserver
+from traits.observers._testing import (
+    call_add_or_remove_notifiers,
+    create_graph,
+    DummyObservable,
+    DummyObserver,
+    DummyNotifier,
+)
+from traits.trait_set_object import TraitSet
+from traits.trait_types import Set
+
+
+def create_observer(**kwargs):
+    """ Convenient function for creating SetItemObserver with default values.
+    """
+    values = dict(
+        notify=True,
+        optional=False,
+    )
+    values.update(kwargs)
```

Comment:
Yet another place where Python 3.9's `|` operator would have been handy ...  (Nothing actionable here; I just wish that the `|` operator for `dict`s had been introduced long ago.)